diff --git a/pcbnew/router/pns_kicad_iface.cpp b/pcbnew/router/pns_kicad_iface.cpp
index 23d8f1295fdb569594cd38365d217d715000768e..9ae2bb5e4e0da58e5528ccf6f04a05e01e979232 100644
--- a/pcbnew/router/pns_kicad_iface.cpp
+++ b/pcbnew/router/pns_kicad_iface.cpp
@@ -1104,6 +1104,10 @@ bool PNS_KICAD_IFACE::IsItemVisible( const PNS::ITEM* aItem )
             && item->ViewGetLOD( item->GetLayer(), m_view ) < m_view->GetScale() )
         return true;
 
+    // Items hidden in the router are not hidden on the board
+    if( m_hiddenItems.find( item ) != m_hiddenItems.end() )
+        return true;
+
     return false;
 }
 
