diff --git a/src/slic3r/GUI/Plater.cpp b/src/slic3r/GUI/Plater.cpp
index 668c5559a..b888ca57c 100644
--- a/src/slic3r/GUI/Plater.cpp
+++ b/src/slic3r/GUI/Plater.cpp
@@ -1160,11 +1160,11 @@ Plater::priv::priv(Plater *q, MainFrame *main_frame)
     panel_sizer = new wxBoxSizer(wxHORIZONTAL);
     panel_sizer->Add(view3D, 1, wxEXPAND | wxALL, 0);
     panel_sizer->Add(preview, 1, wxEXPAND | wxALL, 0);
-    hsizer->Add(panel_sizer, 1, wxEXPAND | wxALL, 0);
+    hsizer->Add(panel_sizer, 3, wxEXPAND | wxALL, 0);
 #else
-    hsizer->Add(notebook, 1, wxEXPAND | wxTOP, 1);
+    hsizer->Add(notebook, 3, wxEXPAND | wxTOP, 1);
 #endif // ENABLE_REMOVE_TABS_FROM_PLATER
-    hsizer->Add(sidebar, 0, wxEXPAND | wxLEFT | wxRIGHT, 0);
+    hsizer->Add(sidebar, 1, wxEXPAND | wxLEFT | wxRIGHT, 0);
     q->SetSizer(hsizer);
 
 //#if ENABLE_REMOVE_TABS_FROM_PLATER
diff --git a/src/slic3r/GUI/Tab.cpp b/src/slic3r/GUI/Tab.cpp
index 41318a854..1a3132c56 100644
--- a/src/slic3r/GUI/Tab.cpp
+++ b/src/slic3r/GUI/Tab.cpp
@@ -189,7 +189,7 @@ void Tab::create_preset_tab()
 
 	//left vertical sizer
 	m_left_sizer = new wxBoxSizer(wxVERTICAL);
-	m_hsizer->Add(m_left_sizer, 0, wxEXPAND | wxLEFT | wxTOP | wxBOTTOM, 3);
+	m_hsizer->Add(m_left_sizer, 1, wxEXPAND | wxLEFT | wxTOP | wxBOTTOM, 3);
 
 	// tree
 	m_treectrl = new wxTreeCtrl(panel, wxID_ANY, wxDefaultPosition, wxSize(185, -1),
@@ -272,7 +272,7 @@ Slic3r::GUI::PageShp Tab::add_options_page(const wxString& title, const std::str
 	PageShp page(new Page(panel, title, icon_idx));
 	page->SetScrollbars(1, 20, 1, 2);
 	page->Hide();
-	m_hsizer->Add(page.get(), 1, wxEXPAND | wxLEFT, 5);
+	m_hsizer->Add(page.get(), 6, wxEXPAND | wxLEFT, 5);
 
     if (!is_extruder_pages)
 		m_pages.push_back(page);
