--- src/slic3r/GUI/Tab.cpp.orig	2019-01-25 19:26:35.968559633 +0100
+++ src/slic3r/GUI/Tab.cpp	2019-01-25 19:26:37.368559717 +0100
@@ -198,7 +198,7 @@
 
 	//left vertical sizer
 	m_left_sizer = new wxBoxSizer(wxVERTICAL);
-	m_hsizer->Add(m_left_sizer, 0, wxEXPAND | wxLEFT | wxTOP | wxBOTTOM, 3);
+	m_hsizer->Add(m_left_sizer, 1, wxEXPAND | wxLEFT | wxTOP | wxBOTTOM, 3);
 
 	// tree
 	m_treectrl = new wxTreeCtrl(panel, wxID_ANY, wxDefaultPosition, wxSize(185, -1), 
@@ -281,7 +281,7 @@
 	PageShp page(new Page(panel, title, icon_idx));
 	page->SetScrollbars(1, 20, 1, 2);
 	page->Hide();
-	m_hsizer->Add(page.get(), 1, wxEXPAND | wxLEFT, 5);
+	m_hsizer->Add(page.get(), 6, wxEXPAND | wxLEFT, 5);
 
     if (!is_extruder_pages) 
 		m_pages.push_back(page);
--- src/slic3r/GUI/Plater.cpp.orig	2019-01-25 19:23:02.457546786 +0100
+++ src/slic3r/GUI/Plater.cpp	2019-01-25 19:24:54.097553503 +0100
@@ -1096,8 +1096,8 @@
     panel_sizer = new wxBoxSizer(wxHORIZONTAL);
     panel_sizer->Add(view3D, 1, wxEXPAND | wxALL, 0);
     panel_sizer->Add(preview, 1, wxEXPAND | wxALL, 0);
-    hsizer->Add(panel_sizer, 1, wxEXPAND | wxALL, 0);
-    hsizer->Add(sidebar, 0, wxEXPAND | wxLEFT | wxRIGHT, 0);
+    hsizer->Add(panel_sizer, 3, wxEXPAND | wxALL, 0);
+    hsizer->Add(sidebar, 1, wxEXPAND | wxLEFT | wxRIGHT, 0);
     q->SetSizer(hsizer);
 
     init_object_menu();
