diff -aur work/src/hydrogen-0.9.6-beta3/CMakeLists.txt hydrogen-0.9.6-beta3/CMakeLists.txt
--- work/src/hydrogen-0.9.6-beta3/CMakeLists.txt	2013-02-21 19:53:51.000000000 +0100
+++ hydrogen-0.9.6-beta3/CMakeLists.txt	2013-10-06 14:53:35.305116743 +0200
@@ -171,8 +171,9 @@
 FIND_HELPER(PORTAUDIO portaudio-2.0 portaudio.h portaudio)
 FIND_HELPER(PORTMIDI portmidi portmidi.h portmidi)
 FIND_HELPER(LASH lash-1.0 lash/lash.h lash)
-FIND_HELPER(LRDF lrdf lrdf.h lrdf)
 FIND_HELPER(RUBBERBAND rubberband RubberBandStretcher.h rubberband)
+# Find LRDF through pkg-config
+pkg_check_modules(LRDF lrdf)
 # CHECK FOR Qt4
 SET(QT_USE_QTNETWORK true)
 SET(QT_USE_QTXML true)
@@ -254,7 +255,7 @@
 COLOR_MESSAGE("${cyan}Usefull extensions${reset}
 -----------------------------------------
 * ${purple}LASH${reset}                         : ${LASH_STATUS}
-* ${purple}LRDF${reset}                         : ${LRDF_STATUS}
+* ${purple}LRDF${reset}                         : ${LRDF_FOUND}
 * ${purple}RUBBERBAND${reset}                   : ${RUBBERBAND_STATUS}
 *                                ${LIBRUBBERBAND_MSG}\n"
 )
diff -aur work/src/hydrogen-0.9.6-beta3/src/core/CMakeLists.txt hydrogen-0.9.6-beta3/src/core/CMakeLists.txt
--- work/src/hydrogen-0.9.6-beta3/src/core/CMakeLists.txt	2013-02-21 19:53:51.000000000 +0100
+++ hydrogen-0.9.6-beta3/src/core/CMakeLists.txt	2013-10-06 14:48:12.170623106 +0200
@@ -21,7 +21,7 @@
     ${COREAUDIO_INCLUDE_DIR}
     ${COREMIDI_INCLUDE_DIR}
     ${LASH_INCLUDE_DIR}
-    ${LRDF_INCLUDE_DIR}
+    ${LRDF_INCLUDE_DIRS}
     ${RUBBERBAND_INCLUDE_DIR}
 )
 
@@ -42,7 +42,7 @@
     ${COREAUDIO_LIBRARIES}
     ${COREMIDI_LIBRARIES}
     ${LASH_LIBRARIES}
-    ${LRDF_LIBRARIES}
+    ${LRDF_LDFLAGS}
     ${RUBBERBAND_LIBRARIES}
 )
 
