--- src/extension/internal/pdfinput/pdf-parser.cpp.orig	2025-01-03 14:24:50.787048669 +0100
+++ src/extension/internal/pdfinput/pdf-parser.cpp	2025-01-03 14:24:51.783048699 +0100
@@ -2392,12 +2392,13 @@
     GBool haveColorKeyMask, haveExplicitMask, haveSoftMask;
     GBool maskInvert;
     GBool maskInterpolate;
+    bool hasAlpha;
     Object obj1, obj2;
     
     // get info from the stream
     bits = 0;
     csMode = streamCSNone;
-    str->getImageParams(&bits, &csMode);
+    str->getImageParams(&bits, &csMode, &hasAlpha);
     
     // get stream dict
     dict = str->getDict();
